internal/services/voiceservices/voice_services_communications_gateway_resource.go

```diff
@@ -0,0 +1,616 @@
+		}
+
+		outputList = append(outputList, output)
+	}
+
+	return outputList
+}
+
+func expandPrimaryRegionPropertiesModel(inputList []PrimaryRegionPropertiesModel) *communicationsgateways.PrimaryRegionProperties {
+	if len(inputList) == 0 {
+		return nil
+	}
+
+	input := &inputList[0]
+	output := communicationsgateways.PrimaryRegionProperties{
+		AllowedMediaSourceAddressPrefixes:     &input.AllowedMediaSourceAddressPrefixes,
+		AllowedSignalingSourceAddressPrefixes: &input.AllowedSignalingSourceAddressPrefixes,
+		EsrpAddresses:                         &input.EsrpAddresses,
+		OperatorAddresses:                     input.OperatorAddresses,
+	}
+
+	return &output
+}
+
+func expandCommunicationsPlatformModel(input []string) []communicationsgateways.CommunicationsPlatform {
+	if len(input) == 0 {
+		return nil
+	}
+
+	var output []communicationsgateways.CommunicationsPlatform
+	for _, v := range input {
+		platform := communicationsgateways.CommunicationsPlatform(v)
+		output = append(output, platform)
+	}
+
+	return output
+}
+
+func flattenServiceRegionPropertiesModel(inputList *[]communicationsgateways.ServiceRegionProperties) []ServiceRegionPropertiesModel {
+	outputList := make([]ServiceRegionPropertiesModel, 0)
+	if inputList == nil {
+		return outputList
+	}
+
+	for _, input := range *inputList {
+		output := ServiceRegionPropertiesModel{
+			Name: input.Name,
+		}
+
+		output.PrimaryRegionProperties = flattenPrimaryRegionPropertiesModel(&input.PrimaryRegionProperties)
+
+		outputList = append(outputList, output)
+	}
+
+	return outputList
+}
+
+func flattenPrimaryRegionPropertiesModel(input *communicationsgateways.PrimaryRegionProperties) []PrimaryRegionPropertiesModel {
+	outputList := make([]PrimaryRegionPropertiesModel, 0)
+	if input == nil {
+		return outputList
+	}
+
+	output := PrimaryRegionPropertiesModel{
+		OperatorAddresses: input.OperatorAddresses,
+	}
+
+	if input.AllowedMediaSourceAddressPrefixes != nil {
+		output.AllowedMediaSourceAddressPrefixes = *input.AllowedMediaSourceAddressPrefixes
+	}
+
+	if input.AllowedSignalingSourceAddressPrefixes != nil {
+		output.AllowedSignalingSourceAddressPrefixes = *input.AllowedSignalingSourceAddressPrefixes
+	}
+
+	if input.EsrpAddresses != nil {
+		output.EsrpAddresses = *input.EsrpAddresses
+	}
+
+	return append(outputList, output)
+}
+
+func flattenCommunicationsPlatformModel(input []communicationsgateways.CommunicationsPlatform) []string {
+	output := make([]string, 0)
+	if len(input) == 0 {
+		return nil
+	}
+
+	for _, v := range input {
+		output = append(output, string(v))
+	}
+
+	return output
+}
+
+func serviceLocations() *schema.Schema {
+	return &schema.Schema{
+		Type:     pluginsdk.TypeSet,
+		Required: true,
+		Elem: &pluginsdk.Resource{
+			Schema: map[string]*pluginsdk.Schema{
+				"name": {
+					Type:         pluginsdk.TypeString,
+					Required:     true,
+					ValidateFunc: validation.StringIsNotEmpty,
+				},
```

Comment:
This is still called `name` this should be called `location`

Reply:
Thanks, the code has been updated. Could you please take another look?